src/toxcore/dht/client.rs

```diff
@@ -125,12 +127,65 @@ impl Client {
         let nat_ping_req = DhtPacket::DhtRequest(DhtRequest::new(&self.precomputed_key.clone(), &friend_pk, &self.pk, payload));
         self.send(nat_ping_req)
     }
+    // get broadcast addresses for host's network interfaces
+    fn get_ipv4_broadcast_addrs() -> Vec<SocketAddr> {
+        let ifs = pnet_datalink::interfaces();
+        let mut res = Vec::new();
+        ifs.iter().for_each(|interface| 
+            interface.ips
+                .iter()
+                .filter_map(|ip| {
+                    match ip.ip() {
+                        IpAddr::V4(ip) => Some(ip.octets()),
```

Comment:
Do you serialize octests to parse them as ip addr again? Oo

Reply:
To get broadcast address.
I will try better way.

Reply:
I am focusing CI build pass

Reply:
Why did you get rid of previous code where you just called broadcast method?

Reply:
Because, I move from ipnetwork crate to ipaddress crate

Reply:
Changing crate is for CI build pass

Reply:
The source pushed is not for merge, just test if it pass CI build

Reply:
ok, but ipnetwork doesn't depend on external libraries so it shouldn't cause any problems. Problem may be in pnet_datalink though.